homeassistant/components/climate/evohome.py

```diff
@@ -279,93 +541,45 @@ def set_operation_mode(self, operation_mode):
+        its children (e.g. Zones, DHW controller).
         """
-        evo_data = self.hass.data[DATA_EVOHOME]
-
+        # should the latest evohome state data be retreived this cycle?
         timeout = datetime.now() + timedelta(seconds=55)
         expired = timeout > self._timers['statusUpdated'] + \
-            timedelta(seconds=evo_data['params'][CONF_SCAN_INTERVAL])
+            timedelta(seconds=self._params[CONF_SCAN_INTERVAL])
 
         if not expired:
             return
 
-        was_available = self._available or \
-            self._timers['statusUpdated'] == datetime.min
-
-        self._update_state_data(evo_data)
-        self._status = evo_data['status']
-
-        if _LOGGER.isEnabledFor(logging.DEBUG):
-            tmp_dict = dict(self._status)
-            if 'zones' in tmp_dict:
-                tmp_dict['zones'] = '...'
-            if 'dhw' in tmp_dict:
-                tmp_dict['dhw'] = '...'
-
-            _LOGGER.debug(
-                "update(%s), self._status = %s",
-                self._id + " [" + self._name + "]",
-                tmp_dict
-            )
-
-        no_recent_updates = self._timers['statusUpdated'] < datetime.now() - \
-            timedelta(seconds=self._params[CONF_SCAN_INTERVAL] * 3.1)
-
-        if no_recent_updates:
-            self._available = False
-            debug_code = EVO_DEBUG_NO_RECENT_UPDATES
-
-        elif not self._status:
-            # unavailable because no status (but how? other than at startup?)
-            self._available = False
-            debug_code = EVO_DEBUG_NO_STATUS
+        # Retreive the latest state data via the client api
+        loc_idx = self._params[CONF_LOCATION_IDX]
 
+        try:
+            self._status.update(
+                self._client.locations[loc_idx].status()[GWS][0][TCS][0])
+        except HTTPError as err:  # check if we've exceeded the api rate limit
+            self._handle_requests_exceptions(err)
         else:
+            self._timers['statusUpdated'] = datetime.now()
             self._available = True
 
-        if not self._available and was_available:
-            # only warn if available went from True to False
-            _LOGGER.warning(
-                "The entity, %s, has become unavailable, debug code is: %s",
-                self._id + " [" + self._name + "]",
-                debug_code
-            )
+        _LOGGER.debug(
+            "_update_state_data(): self._status = %s",
+            self._status
+        )
 
-        elif self._available and not was_available:
-            # this isn't the first re-available (e.g. _after_ STARTUP)
-            _LOGGER.debug(
-                "The entity, %s, has become available",
-                self._id + " [" + self._name + "]"
-            )
+        # inform the child devices that state data has been updated
+        pkt = {'sender': 'controller', 'signal': 'refresh', 'to': EVO_CHILD}
+        async_dispatcher_send(self.hass, DISPATCHER_EVOHOME, pkt)
```

Comment:
Use sync version. We're in sync context so have to use the sync api of home assistant.